lib/token.js

```diff
@@ -247,7 +250,9 @@ function convertOAuthParamsToQueryParams(oauthParams) {
     'display': oauthParams.display,
     'sessionToken': oauthParams.sessionToken,
     'idp': oauthParams.idp,
-    'max_age': oauthParams.maxAge
+    'max_age': oauthParams.maxAge,
+    'code_challenge': oauthParams.codeChallenge,
+    'code_challenge_method': oauthParams.codeChallenge ? 'S256' : null
```

Comment:
Ideally this value will be pulled from the `.well-known/openid-configuration` document to maintain future compatibility. A list of available options are given under the key `code_challenge_method`.

Reply:
I am not sure how to do this. Is the '.well-known/...' something I can include as a node module?

Reply:
It's a url - .well-known/(whatever) is a convention to allow an oauth server to be queryable regarding various endpoints. by providing some hardcoded ones that are..."well-known".

Reply:
Correct - see an example [here](https://example.oktapreview.com/oauth2/default/.well-known/openid-configuration).

To make this easy, we have a method called `getWellKnown` that fetches the discovery document. Ideally we will parse the `code_challenge_method` options then if the `oauthParams.codeChallenge` matches our approved values, we will continue with the request.

Reply:
Added this logic